front/app/hooks/files/useSyncFiles.ts

```diff
@@ -0,0 +1,151 @@
+import { useCallback } from 'react';
+
+import { isString } from 'lodash-es';
+
+import useAddPhaseFile from 'api/phase_files/useAddPhaseFile';
+import useDeletePhaseFile from 'api/phase_files/useDeletePhaseFile';
+import useUpdatePhaseFile from 'api/phase_files/useUpdatePhaseFile';
+import useAddProjectFile from 'api/project_files/useAddProjectFile';
+import useDeleteProjectFile from 'api/project_files/useDeleteProjectFile';
+import useUpdateProjectFile from 'api/project_files/useUpdateProjectFile';
+
+import { SyncFilesArguments } from './types';
+
+export function useSyncFiles() {
+  // ***** Project File Hooks *****
+  const { mutateAsync: addProjectFile } = useAddProjectFile();
+  const { mutateAsync: updateProjectFile } = useUpdateProjectFile();
+  const { mutateAsync: deleteProjectFile } = useDeleteProjectFile();
+
+  // ***** Phase File Hooks *****
+  const { mutateAsync: addPhaseFile } = useAddPhaseFile();
+  const { mutateAsync: deletePhaseFile } = useDeletePhaseFile();
+  const { mutateAsync: updatePhaseFile } = useUpdatePhaseFile();
+
```

Comment:
Since Projects/Phases use slightly different file types still, this hook just essentially groups together all the file handling for Phases/Projects (and in my next PR - Folders too) but still handles them separately using conditionals (below).

After that, this hook + the types can be further cleaned up though, and this makes it a bit easier since it's all in the same hook now.